catalog/transformer/catalog-transformer-pdf/src/main/java/ddf/catalog/transformer/input/pdf/PdfInputTransformer.java

```diff
@@ -81,26 +63,102 @@
     private static final FastDateFormat DATE_FORMAT = FastDateFormat.getInstance(
             "yyyy-MM-dd'T'HH:mm:ssZZ");
 
-    private Map<ServiceReference, ContentMetadataExtractor> contentMetadataExtractors =
-            Collections.synchronizedMap(new TreeMap<>(new ServiceComparator()));
+    private final CheckedFunction<InputStream, PDDocument> inputStreamPDDocumentFunction;
+
+    private final CheckedFunction<PDDocument, String> geoParserFunction;
+
+    private final CheckedFunction<PDDocument, byte[]> thumbnailFunction;
+
+    private MetacardType metacardType;
+
+    private boolean usePdfTitleAsTitle;
+
+    /**
+     * @param metacardType       must be non-null
+     * @param usePdfTitleAsTitle must be non-null
+     */
+    public PdfInputTransformer(MetacardType metacardType, Boolean usePdfTitleAsTitle) {
+        this(metacardType,
+                usePdfTitleAsTitle,
+                PDDocument::load,
+                GEO_PDF_PARSER::getWktFromPDF,
+                PdfInputTransformer::generatePdfThumbnail);
+    }
+
+    /**
+     * @param metacardType                  must be non-null
```

Comment:
Can we add a note here saying that this constructor is _only_ for unit testing? We could also get rid of the `notNull` checks since this is not meant for production use.